src/GraphQL/Execution/ExecutionContext.cs

```diff
@@ -60,18 +60,21 @@ public class ExecutionContext : IExecutionContext
         /// <inheritdoc/>
         public IServiceProvider RequestServices { get; set; }
 
-        private readonly List<Array> _trackedArrays = new List<Array>();
-
-        internal void TrackArray(Array array)
+        /// <inheritdoc/>
+        public TElement[] Rent<TElement>(int minimumLength)
         {
-            if (array == null)
-                throw new ArgumentNullException(nameof(array));
-
+            var array = System.Buffers.ArrayPool<TElement>.Shared.Rent(minimumLength);
             lock (_trackedArrays)
                 _trackedArrays.Add(array);
+            return array;
         }
 
-        internal void ReturnArrays()
+        private readonly List<Array> _trackedArrays = new List<Array>();
+
+        /// <summary>
+        /// Releases any rented arrays back to the backing memory pool.
+        /// </summary>
+        public void Dispose()
```

Comment:
It is extremely important that ANY implementation of `IDisposable` be able to be called multiple times.  Doing so in this instance will return the same arrays to the pool again.  That is why it is very important to clear `_trackedArrays`.  It has nothing to do with clearing the variables or not.

If you'd rather implement a pattern where there's an `_disposed` property and `if (_disposed) return;` and etc, be my guest.  But you must anticipate that `Dispose` may be called multiple times.

Reply:
🤦 Done. I decided to clear the context completely. This is a relatively large object with 17 properties. Let's apply pooling to it.